corehq/apps/receiverwrapper/tests/test_submissions.py

```diff
@@ -118,6 +122,97 @@ def test_submit_deprecated_form(self):
         self.assertEqual(self.couch_user.get_id, form.history[0].user)
 
 
+@patch('corehq.apps.receiverwrapper.views.domain_requires_auth', return_value=True)
+class NoAuthSubmissionTest(BaseSubmissionTest):
+    def setUp(self):
+        super(NoAuthSubmissionTest, self).setUp()
+        self.url = self.url + '?authtype=noauth'
+        # skip any authorization
+        self.client = Client()
+
+    def test_successful_processing_for_demo_user_form(self, *_):
+        response = self._submit('demo_mode_simple_form.xml', url=self.url)
+        self.assertTrue('X-CommCareHQ-FormID' in response, 'Demo user ID form not processed in demo mode')
+
+    def test_ignore_all_non_demo_user_submissions(self, *_):
+        response = self._submit('simple_form.xml', url=self.url)
+        self.assertEqual(response.status_code, 403)
+
+
+@patch('corehq.apps.receiverwrapper.views.domain_requires_auth', return_value=True)
+class DefaultDemoModeSubmissionTest(BaseSubmissionTest):
+    """
+        Demo Mode means the request is being sent with param submit_mode=demo and authtype as noauth
+        The user id in the form is expected to be DEMO_USER_ID
+        Only forms submitted with user ID as demo_user are processed.
+    """
+    def setUp(self):
+        super(DefaultDemoModeSubmissionTest, self).setUp()
+        self.domain.secure_submissions = True
+        self.domain.save()
+        self.url = self.url + '?submit_mode=demo&authtype=noauth'
+        # skip any authorization
+        self.client = Client()
+
+    def test_successful_processing_for_demo_user_form(self, *_):
+        response = self._submit('demo_mode_simple_form.xml', url=self.url)
+        self.assertTrue('X-CommCareHQ-FormID' in response, 'Demo user ID form not processed in demo mode')
+
+    def test_ignore_all_non_demo_user_submissions(self, *_):
+        response = self._submit('simple_form.xml', url=self.url)
+        self.assertFalse('X-CommCareHQ-FormID' in response, 'Non Demo ID form processed in demo mode')
+
+
+@patch('corehq.apps.receiverwrapper.views.domain_requires_auth', return_value=True)
+class PracticeMobileWorkerSubmissionTest(BaseSubmissionTest):
+    """
+    Just like demo mode, the request is sent with param submit_mode=demo and authtype=noauth
+    but the userID in the form is expected to be the user ID of the practice mobile worker
+    """
+    def setUp(self):
+        super(PracticeMobileWorkerSubmissionTest, self).setUp()
+        self.url = self.url + '?submit_mode=demo&authtype=noauth'
+        self.domain.secure_submissions = True
```

Comment:
I don't think this is necessary since you're patching `domain_requires_auth`

Reply:
Ah yes, forgot to remove that after adding the patch, thanks